src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+    if (transformContext == null) {
+      transformContext = TransformContext()
+    }
+
+    transformContext = transformContext.copy(
+      siteSettings = delta.siteSettings.getOrElse(transformContext.siteSettings),
+      langToWatchlist = delta.langToWatchlist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.langToWatchlist
+      },
+      blacklist = delta.blacklist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.blacklist
+      },
+      locationsExtractorFactory = delta.locationsExtractorFactory match {
+        case Some(factory) => sparkContext.broadcast(factory)
+        case None => transformContext.locationsExtractorFactory
+      },
+      imageAnalyzer = delta.imageAnalyzer.getOrElse(transformContext.imageAnalyzer),
+      languageDetector = delta.languageDetector.getOrElse(transformContext.languageDetector),
+      sentimentDetectorAuth = delta.sentimentDetectorAuth.getOrElse(transformContext.sentimentDetectorAuth)
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided site settings.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithSettings(siteSettings: SiteSettings, delta: Delta = Delta()): Delta = {
+    // Note that parameters are call-by-name
+    def updatedField[T](isDirty: => Boolean, newVal: => T): Option[T] = {
+      // Invoke 'isDirty' only if transform context is not null
+      if (transformContext == null || isDirty)
+        Some(newVal)
+      else
+        None
+    }
+
+    delta.copy(
+      siteSettings = Some(siteSettings),
+      locationsExtractorFactory = updatedField(
+        siteSettings.geofence != transformContext.siteSettings.geofence,
+        new LocationsExtractorFactory(featureServiceClient, siteSettings.geofence).buildLookup()
+      ),
+      imageAnalyzer = updatedField(
+        siteSettings.cogVisionSvcToken != transformContext.siteSettings.cogVisionSvcToken,
+        new ImageAnalyzer(ImageAnalysisAuth(siteSettings.cogVisionSvcToken), featureServiceClient)
+      ),
+      languageDetector = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        new LanguageDetector(LanguageDetectorAuth(siteSettings.translationSvcToken))
+      ),
+      sentimentDetectorAuth = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        SentimentDetectorAuth(siteSettings.translationSvcToken)
+      )
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided watchlist.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithWatchlist(langToWatchlist: Map[String, List[String]], delta: Delta = Delta()) = {
+    delta.copy(
+      langToWatchlist = Some(langToWatchlist)
```

Comment:
This only lets us replace the entire watchlist in one go. Via the dashboard we'll often be adding single terms, so I wonder if this should instead update the watchlist terms instead of replacing them, so that we don't have to fetch the entire watchlist from the DB. More generally, whenever a message arrives, with the current approach we'll have to fetch the new values from the DB -- why not just put the delta into the queue message directly? Sure, message sizes are limited, but will we really hit them if all we do is update single words? Why refetch data that we already have available :).